lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/auth/OAuthClientDaoImplTest.java

```diff
@@ -476,4 +494,100 @@ public void testIsOauthClientVerified() throws Exception {
 		
 	}
 
+	private OAuthRefreshTokenInformation createRefreshToken(String clientId, String userId, Date authorizedOn, Date lastUsed) {
+		OAuthRefreshTokenInformation token = new OAuthRefreshTokenInformation();
+		token.setName(UUID.randomUUID().toString());
+		token.setClientId(clientId);
+		token.setPrincipalId(userId);
+		token.setScopes(Collections.singletonList(OAuthScope.view));
+		token.setClaims(new OIDCClaimsRequest());
+		token.setAuthorizedOn(authorizedOn);
+		token.setLastUsed(lastUsed);
+		token.setModifiedOn(new Date());
+		token.setEtag(UUID.randomUUID().toString());
+		return oauthRefreshTokenDao.createRefreshToken("abcdef", token);
+	}
+
+	@Test
+	public void testGetAuthorizedClients() {
+		Long userId = BOOTSTRAP_PRINCIPAL.THE_ADMIN_USER.getPrincipalId();
+
+		String client1 = createSectorIdentifierAndClient();
+		String client2 = createSectorIdentifierAndClient(SECOND_SECTOR_IDENTIFIER, userId, "App 2");
+
+		final Date client1ExpectedAuthzOn = new Date(System.currentTimeMillis() - ONE_YEAR_MILLIS * 5);// Earliest authorized on date
+		final Date client1ExpectedLastUsed = new Date(System.currentTimeMillis() + ONE_YEAR_MILLIS * 5);// Latest last used date
+		// Valid tokens for client 1
+		createRefreshToken(client1, userId.toString(), client1ExpectedAuthzOn, new Date());
+		createRefreshToken(client1, userId.toString(), new Date(), new Date());
+		createRefreshToken(client1, userId.toString(), new Date(), client1ExpectedLastUsed);
+
+		// Expired token for client 1. The dates are set up such that the result "authorizedOn" will not match if this token is not filtered.
+		createRefreshToken(client1, userId.toString(), new Date(System.currentTimeMillis() - ONE_YEAR_MILLIS * 10), new Date(System.currentTimeMillis() - TWO_YEARS_MILLIS));
+
+		// Valid token for client 2
+		final Date client2ExpectedAuthzOn = new Date(System.currentTimeMillis() - 542326L);
+		final Date client2ExpectedLastUsed = new Date(System.currentTimeMillis() - 54232L);
+		createRefreshToken(client2, userId.toString(), client2ExpectedAuthzOn, client2ExpectedLastUsed);
+
+		// Call under test
+		OAuthClientAuthorizationHistoryList results = oauthClientDao.getAuthorizedClientHistory(userId.toString(), null, ONE_YEAR_DAYS);
+
+		assertEquals(2, results.getResults().size());
+
+		for (OAuthClientAuthorizationHistory result : results.getResults()) {
+			if (result.getClient().getClient_id().equals(client1)) {
+				assertEquals(client1ExpectedAuthzOn, result.getAuthorizedOn());
+				assertEquals(client1ExpectedLastUsed, result.getLastUsed());
+			}
+			if (result.getClient().getClient_id().equals(client2)) {
+				assertEquals(client2ExpectedAuthzOn, result.getAuthorizedOn());
+				assertEquals(client2ExpectedLastUsed, result.getLastUsed());
+			}
+		}
+
```

Comment:
OAuthClientHistory actualHistory1 = null;
OAuthClientHistory actualHistory2 = null;
for (OAuthClientAuthorizationHistory result : results.getResults()) {
if (result.getClient().getClient_id().equals(client1)) {
actualHistory1 = result;
} else if (result.getClient().getClient_id().equals(client2)) {
actualHistory2 = result;
} else {
fail();
}
}
assertEquals(client1ExpectedAuthzOn, actualHistory1.getAuthorizedOn());
// + 3 more assertions